### PR TITLE
Handle Aggregate Values

### LINKED ISSUE #75: Improve Handling of Aggregates in Open MCT
YAMCS supports "Aggregate" parameter types that associate multiple parameter values with a single timestamp. 

eg. here's a screenshot of the YAMCS interface for an aggregate telemetry point (taken from [YAMCS-Quickstart](https://github.com/yamcs/quickstart))

Screen Shot 2021-08-19 at 4 41 27 PM

Open MCT supports this concept at a platform level, eg. the SineWaveGenerator produces telemetry data objects with both sin and cos attributes associated with a single timestamp. However currently the **Open MCT-YAMCS** telemetry adapters do not. Instead they only support a single "value" attribute per telemetry datum so the telemetry datum objects that they produce look something like this

```JavaScript
{
    id: "~myproject~CCSDS_Packet_Sequence.Count", 
    timestamp: "2021-08-19T23:13:12.812Z",
    value: 39
}
```
So, although the Open MCT-YAMCS plugin does handle YAMCS' aggregate parameter types, but it does so by showing the aggregate parameter as a folder in the tree, and exploding its members out into separate telemetry objects. The same aggregate parameter from the screenshot above is represented in Open MCT like so:

Screen Shot 2021-08-19 at 4 44 12 PM

This is fine, but there are cases (such as spectral plots) where it's useful to be able to treat the aggregate as a single telemetry point where each aggregate member is an attribute of the telemetry datum. So, instead of a single value attribute as in the previous example, the resulting telemetry datum might look something like this:

```JavaScript
{
    id: "~myproject~CCSDS_Packet_Sequence", 
    timestamp: "2021-08-19T23:13:12.812Z",
    groupFlags: 3,
    count: 39
}
```

As mentioned previously, Open MCT handles this type of data just fine with the right telemetry metadata (see the Sine Wave Generator as an example).

So I think the right approach for handling Aggregate telemetry types is to:
1. Create a new "aggregate" object type that we will use to represent aggregate parameters. It will be a telemetry point itself (unlike the current folder representation), so views will be able to subscribe to it. eg. if you dragged an aggregate object into a table you would see a column for each of its members. eg. ch1, ch2, ch3, ch4...etc. in the previous above.
2. Modify the Open MCT-YAMCS plugin telemetry metadata so that instead of just "timestamp" and "value", aggregate objects have metadata for all of their members.
3. The available views for an "aggregate" object type should be either the AutoFlowTabular, the LadTable, or both? I'll let @charlesh88 make the final decision. This will allow users to preview telemetry for all of the members of the aggregate without even having to build a display.

Now, here's where it gets really interesting. Aggregate objects should also have composition! They should still expose each of their members as individual telemetry points, just like we currently do! This is for backwards compatibility, but also to give users the option of handling aggregate members individually or all together.

### PR DESCRIPTION
**Pull request by jvigliotta**
Formatting metadata for aggregate values and updating providers to work with aggregates. Also refactored historical provider as it was getting a little overcomplicated.

closes #75

### Author Checklist
Check | Passed
-- | --
Changes address original issue? | Y
Unit tests included and/or updated with changes? | N/A
Command line build passes? | Y
Changes have been smoke-tested? | Y
Testing instructions included? | Y

### CODE REVIEW

File: src/providers/object-provider.js [hunk 1]
```diff
@@ -300,68 +300,121 @@ export default class YamcsObjectProvider {
...
+                key = parentKey + '.' + key;
+                name = parentKey.split('.').shift() + ' ' + name;
+            }
+
+            if (!this.isAggregate(member)) {
+                formatted.push({
+                    key,
+                    name,
+                    hints: {
+                        range: rangeHint++
```
Comment (shefalijoshi) [inline src/providers/object-provider.js:398]: Can we really assume that each member of an aggregate is a range? I think we need something more from yamcs telling us if the member is a domain or a range, no?

Comment (jvigliotta) [reply src/providers/object-provider.js:398]: yeah, I don't think assigning each one an incrementing range is correct... I'm not sure how or if we assign them hints

**Comment by shefalijoshi** [comment]
@jvigliotta On this line: https://github.com/akhenry/openmct-yamcs/pull/77/files#diff-58d0a7ee67c74185338cf7e69ad5f3ce1d68be3142fa93edd6f5f6af40f9c22eR82

Can we add a check that `this.subscriptionsById[id]` exists before we invoke .call on it?

File: src/providers/object-provider.js (hunk 1 above)
Comment (shefalijoshi) [reply src/providers/object-provider.js:399]: This is fine as per our conversation with Andrew. Let's keep it.

**Pull request merged**